olive/passes/olive_pass.py

```diff
@@ -43,25 +43,37 @@ class Pass(AutoConfigClass):
     # True if pass configuration requires user script for non-local host support
     _requires_user_script: bool = False
 
-    def __init__(
-        self, config: Optional[Union[Dict[str, Any], PassConfigBase]] = None, disable_search: Optional[bool] = False
-    ):
-        """
-        Initialize the pass.
-        disable_search: If False, use default search parameters, if any, for parameters that are not specified
-        in the config. Only applies when the config is a dictionary.
-        """
-        self._config_class = self.get_config_class(disable_search)
-        self._config = validate_config(config, PassConfigBase, self._config_class)
-        self._config = self._config.dict()
-        self._config = self._resolve_defaults(self._config)
+    @classmethod
+    def __init_subclass__(cls, **kwargs) -> None:
+        """Register the metric."""
+        super().__init_subclass__(**kwargs)
+        if not inspect.isabstract(cls):
+            name = cls.name if cls.name is not None else cls.__name__.lower()
```

Comment:
`name` was a nullable class attribute for `AutoConfigClass`. We could just use `cls.__name__.lower()` here directly.

Reply:
That is also the actual source of the failure here
![image](https://user-images.githubusercontent.com/94929125/233252844-6a0e4f16-bc42-4146-a020-51a3e1e0bbd7.png)

Reply:
yeah, I already realized it and change to use cls.__name__ instead.